### PR TITLE
Unblock perfdash upgrade

### LINKED ISSUE #1792: Upgrade of perfdash to 2.33 is blocked
We needed to revert 2.33 perfdash upgrade in https://github.com/kubernetes/perf-tests/pull/1784, because there is not enough resources (memory) in the aaa cluster. 

To unblock that, https://github.com/kubernetes/k8s.io/pull/1604 needs to happen

### PR DESCRIPTION
**Pull request by VivekThrivikraman-est**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This PR is for unblocking the perfdash upgrade.

**Which issue(s) this PR fixes**:
Fixes #1792 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: perfdash/deployment.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ spec:
       - name: perfdash
         # Version 2.33 consumes > 32GiB of memory.
         # TODO: Introduce 2.34 version which conditionally disables functionality added in 2.32 and disable it in this perf-dash instance.
```
Comment (mm4tt) [inline perfdash/deployment.yaml:19]: Let's remove this TODO

**Comment by VivekThrivikraman-est** [comment]
@mm4tt

File: perfdash/metrics-downloader.go [hunk 2]
```diff
@@ -42,15 +42,17 @@ type DownloaderOptions struct {
 
 // Downloader that gets data about results from a storage service (GCS) repository.
 type Downloader struct {
-	MetricsBkt MetricsBucket
-	Options    *DownloaderOptions
+	MetricsBkt           MetricsBucket
+	Options              *DownloaderOptions
+	allowParsersAllTests *bool
```
Comment (mm4tt) [inline perfdash/metrics-downloader.go:47]: Let's rename to allowParsersForAllTests and make it bool not *bool

**Review by mm4tt — COMMENTED** [review]
Looks good, just two minor comments

File: perfdash/deployment.yaml [hunk 3]
```diff
@@ -16,7 +16,6 @@ spec:
       containers:
       - name: perfdash
         # Version 2.33 consumes > 32GiB of memory.
```
Comment (mm4tt) [inline perfdash/deployment.yaml:18]: This one as well :)

File: perfdash/metrics-downloader.go [hunk 4]
```diff
@@ -42,15 +42,17 @@ type DownloaderOptions struct {
...
-	MetricsBkt MetricsBucket
-	Options    *DownloaderOptions
+	MetricsBkt              MetricsBucket
+	Options                 *DownloaderOptions
+	allowParsersForAllTests bool
 }
 
 // NewDownloader creates a new Downloader.
-func NewDownloader(opt *DownloaderOptions, bkt MetricsBucket) *Downloader {
+func NewDownloader(opt *DownloaderOptions, bkt MetricsBucket, allowAllParsers *bool) *Downloader {
```
Comment (mm4tt) [inline perfdash/metrics-downloader.go:51]: This should be also bool, it doesn't make sense to allow *bool here, what should we do if caller provides nil?

**Review by mm4tt — COMMENTED** [review]
Please address the remaining comment and squash all commits together and it'll be good to merge

**Comment by mm4tt** [comment]
Adding @jprzychodzen, the current scalability oncall.

Kuba, could you push the new perf-dash version once this PR merges?

**Comment by jprzychodzen** [comment]
Sure, I will push new perf-dash.

Thanks for this change!

**Comment by jprzychodzen** [comment]
FYI - new perf-dash is up and running.

**Pull request merged**